RedHatInsights/curiosity-frontend
# feat(inventorySubscriptions) ent-3352 activate subscription table

## What's included
- Removed dotenv check for product views (RHEL / Openshift) for the subscription inventory table
- confirmed API endpoint is valid
- modified the inventory columns per the storyboard

### Notes
- Table is not currently sortable by title or by the cores/sockets; I tried to update but was getting a 404 back from the API, and it was noted in the story in Jira that "not all inventory column features may be available initially and that it's known to UX/D"
- Also the UOM filtering functionality on the API side is still pending, but with these changes, the UI sorting in Openshift should function correctly.
- Lastly, there is a scenario when filtering by Service Level on RHEL, descending, where UOM is returned as null from the API, resulting in an error in the header; hence why the header config for that has a default value of 'sockets'. 
<img width="1489" alt="Screen Shot 2021-09-16 at 10 48 16 AM" src="https://user-images.githubusercontent.com/4838984/133634206-f41244c6-716e-48e1-b4c1-12d8ca303881.png">

## How to test
Proxy run check
update the NPM packages with `$ yarn`
make sure Docker is running, plus on network, then
`$ yarn start:proxy`
navigate to the RHEL product view.
click "Current subscriptions" tab
confirm that data is showing properly
confirm that data is sortable by clicking Service Level, Quantity, and Next Renewal
navigate to the OpenShift product view
click "Current subscriptions" tab
confirm that data is showing properly
confirm that data is sortable by clicking Service Level, Quantity, and Next Renewal
confirm that cores / sockets header and data (once the API has been updated for this) are updated appropriately when clicking the Cores/Sockets select above

## Example
<img width="1490" alt="Screen Shot 2021-09-16 at 10 44 32 AM" src="https://user-images.githubusercontent.com/4838984/133634772-fca40f21-a73a-4086-a9af-0c4f73ec8e2b.png">

<img width="1479" alt="Screen Shot 2021-09-16 at 10 53 31 AM" src="https://user-images.githubusercontent.com/4838984/133634935-f31beafa-0b6d-4176-a520-8d6e46494db1.png">

...

## Updates issue/story
[ENT-3352](https://issues.redhat.com/browse/ENT-3352)
